build-system/tasks/e2e/describes-e2e.js

```diff
@@ -483,43 +483,49 @@ class EndToEndFixture {
    * @param {number} retries
    */
   async setup(env, browserName, retries = 0) {
+    const {testUrl, experiments = [], initialRect, deviceName} = this.spec;
+    const config = getConfig();
+    let driver;
     try {
-      const {testUrl, experiments = [], initialRect, deviceName} = this.spec;
-      const config = getConfig();
-      const controller = await getController(config, browserName, deviceName);
-      const ampDriver = new AmpDriver(controller);
-      const requestBank = new RequestBankE2E(REQUESTBANK_URL_PREFIX, 'e2e');
-      env.controller = controller;
-      env.ampDriver = ampDriver;
-      env.requestBank = requestBank;
-      const {environment} = env;
-
-      installBrowserAssertions(controller.networkLogger);
-
-      const url = new URL(testUrl);
-      if (experiments.length > 0) {
-        if (environment.includes('inabox')) {
-          // inabox experiments are toggled at server side using <meta> tag
-          url.searchParams.set('exp', experiments.join(','));
-        } else {
-          // AMP doc experiments are toggled via cookies
-          await toggleExperiments(ampDriver, url.href, experiments);
-        }
-      }
-
-      if (initialRect) {
-        const {width, height} = initialRect;
-        await controller.setWindowRect({width, height});
-      }
-
-      await ampDriver.navigateToEnvironment(environment, url.href);
+      driver = await getDriver(config, browserName, deviceName);
     } catch (ex) {
       if (retries > 0) {
         await this.setup(env, browserName, --retries);
       } else {
         throw ex;
       }
     }
+
+    const controller =
+      config.engine == 'puppeteer'
+        ? new PuppeteerController(driver)
+        : new SeleniumWebDriverController(driver);
```

Comment:
Optional: Instead of hard-coding strings for the value of `engine`, maybe make it an enum?

Reply:
Done.